src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs

```diff
@@ -0,0 +1,296 @@
+
+
+        /// <summary>
+        /// Is the recomputeAll button enabled in the UI. Users should not be able to force a 
+        /// reset of the engine and re-execution of the graph if one is still ongoing. This causes...trouble.
+        /// Source: TuneUp https://github.com/DynamoDS/TuneUp
+        /// </summary>
+        public bool IsRecomputeEnabled
+        {
+            get => isRecomputeEnabled;
+            private set
+            {
+                if (isRecomputeEnabled != value)
+                {
+                    isRecomputeEnabled = value;
+                    RaisePropertyChanged(nameof(IsRecomputeEnabled));
+                }
+            }
+        }
+        #endregion
+
+        #region Constructor
+        /// <summary>
+        /// Constructor
+        /// Initialize the ViewLoadedParameters that contain most of the interesting stuff 
+        /// Subscribe to the Workspace Changed and Cleared Events
+        /// Establish the Current Workspace
+        /// </summary>
+        /// <param name="p"></param>
+        public GraphNodeManagerViewModel(ViewLoadedParams p)
+        {
+            this.viewLoadedParams = p;
+
+            p.CurrentWorkspaceChanged += OnCurrentWorkspaceChanged;
+            p.CurrentWorkspaceCleared += OnCurrentWorkspaceCleared;
+            
+            if (p.CurrentWorkspaceModel is HomeWorkspaceModel)
+            {
+                CurrentWorkspace = p.CurrentWorkspaceModel as HomeWorkspaceModel;
+            }
+
+            InitializeFilters();
+        }
+
+        private void InitializeFilters()
+        {
+            FilterItems.Add(new FilterViewModel(){Name = "Empty List"});
+            FilterItems.Add(new FilterViewModel(){Name = "Error"});
+            FilterItems.Add(new FilterViewModel(){Name = "Frozen"});
+            FilterItems.Add(new FilterViewModel(){Name = "Function"});
+            FilterItems.Add(new FilterViewModel(){Name = "Information"});
+            FilterItems.Add(new FilterViewModel(){Name = "Is Input"});
+            FilterItems.Add(new FilterViewModel(){Name = "Is Output"});
+            FilterItems.Add(new FilterViewModel(){Name = "Null"});
+            FilterItems.Add(new FilterViewModel(){Name = "Warning"});
+            FilterItems.Add(new FilterViewModel(){Name = "Preview off"});
+        }
+
+        #endregion
+
+        #region Node Methods
+        /// <summary>
+        /// The main method to set the Node Collection
+        /// Will be triggered in two occasions:
+        /// - a new workspace is established
+        /// - on Run (or after Run has finished?)
+        /// </summary>
+        private void ResetNodes()
+        {
+            if (CurrentWorkspace == null)
+            {
+                return;
+            }
+            nodeDictionary.Clear();
+            Nodes.Clear();
+            foreach (var node in CurrentWorkspace.Nodes)
+            {
+                var profiledNode = new NodeViewModel(node);
```

Comment:
left over  from tuneup?

Reply:
sure, but `profiledNode` does not seem the appropriate name for this variable.

Reply:
Done and done.